app/Jobs/ApplicationDeploymentJob.php

```
@@ -938,6 +949,37 @@ private function deploy_nixpacks_buildpack()
+
+        // Save build-time .env file BEFORE the build
+        $this->save_buildtime_environment_variables();
+
+        $this->generate_build_env_variables();
+        $this->build_railpack_image();
+
+        // Save runtime environment variables AFTER the build
+        $this->save_runtime_environment_variables();
+        $this->push_to_docker_registry();
+        $this->rolling_update();
+    }
```

PHPMD (2.15.0)

[error] The method deploy_railpack_buildpack is not named in camelCase. (undefined)

(CamelCaseMethodName)